### DISCUSSION TITLE
InteractiveAuthenticationProvider not producing auth form/browser in VSTO

### DISCUSSION THREAD

**Discussion by Retcorp** [pnp/pnpcore#524] [Q&A]
I am using the PnP InteractiveAuthenticationProvider to get a context using the code below. When I put it in a Windows app (.Net Framework 4.72) on a form, it pops up an interactive auth window to allow me to select the user to use. When I put it in a separate new project that is a desktop app (.Net 5), it pops up a browser with an appropriate auth page. These both work nicely (but are different to each other for some reason when utilising the same code), and once the auth is confirmed, they both return a PnPContext that I can happily make SPO calls using. Unfortunately, when I try this identical code (with identical variables) in a VSTO plugin for outlook (The project I actually need to use this mechanism in), there is no popup and it returns me an error of:

"Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host."

Which I think means that it can't find anything listening on the endpoint it is calling? It's bound to be something daft that I'm doing, but some pointers would be fantastic. Here is the code:

```
var host = Host.CreateDefaultBuilder()

.ConfigureServices((hostingContext, services) =>
{
	//Add the PnP Core SDK library services
	services.AddPnPCore(options =>
	{
		options.DefaultAuthenticationProvider = new InteractiveAuthenticationProvider(
						AppClientID,
						tenantId,
						new Uri("http://localhost"));
	});
})

// Add services to the container
.Build();

await host.StartAsync();

// Connect to SharePoint
using (var scope = host.Services.CreateScope())
{
	// Obtain a PnP Context factory
	var pnpContextFactory = scope.ServiceProvider.GetRequiredService<IPnPContextFactory>();

	// Use the PnP Context factory to get a PnPContext for the given configuration
	using (var context = await pnpContextFactory.CreateAsync(new Uri(webUrl)))
	{
		return context;
	}
}
```

**Comment by jansenbe** [comment]
@Retcorp : that's weird, I'm not a VSTO developer so hard for me to provide an immediate solution...I do know that this scenario also works in WPF applications (see https://github.com/pnp/pnpcore/tree/dev/samples/Demo.WPF as our sample). 

The error itself might indicate a TLS issue, if you're on an older .NET Framework version you might default to TLS 1.0 which has been deprecated in Microsoft 365. Can you try your plugin compiled on a recent .NET Framework version or on .NET Core? Alternatively try forcing TLS by setting it via `ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls11 | SecurityProtocolType.Tls12;`
